### ISSUE TITLE
Time bigger than 24:00

### ISSUE THREAD

**Issue by ddimitrioglo** [moment/luxon#915]
**The bug**

The `.toFormat()` method gave the `24:30` time for some reason.

**To Reproduce**

I could not reproduce it but I see in production logs that on `2021-03-30T21:53:05.354+0000` the following line

`DateTime.fromISO('2021-03-30T23:30:00.000+0000').toFormat("y-LL-dd'T'TT'.'SSSZZZ")`

produced the following result `2021-03-31T24:30:00.000-0000`

**Actual vs Expected behavior**

Actual: `2021-03-31T24:30:00.000-0000`
Expected: `2021-03-31T00:30:00.000-0000`

**Machine info**

- OS: Linux 4.14.214-160.339.amzn2.x86_64 x86_64 GNU/Linux

**Comment by Yaro77** [comment]
On Chrome 89 (Windows 10) the output is
> luxon.DateTime.fromISO('2021-03-30T23:30:00.000+0000').toFormat("y-LL-dd'T'TT'.'SSSZZZ")
< "2021-03-31T02:30:00.000+0300"

**Comment by icambron** [comment]
Interesting. Sounds like a bug

**Comment by icambron** [comment]
oops, didn't mean to close

**Comment by icambron** [comment]
I can't reproduce this

**Comment by ramos-ph** [comment]
I were able to reproduce this. Since I'm from Brazil, my timezone GMT-3. So I tried with a different input:

```js
date = DateTime.fromISO("2021-03-30T03:30:00.000+0000").toFormat(
  "y-LL-dd'T'TT'.'SSSZZZ"
);

console.log(date);
// > 2021-03-30T24:30:00.000-0300
```

**Comment by ramos-ph** [comment]
@icambron @ddimitrioglo I found out what the issue is. It's caused by `Intl.DateTimeFormat` and not by luxon itself.
I ran into the inner date formatters and found out that under the hood the we use `DateTimeFormat`.

Here's what I've tested:
```js
date = new Date(Date.UTC(2020, 11, 20, 3, 23, 16, 738));

console.log(
  new Intl.DateTimeFormat("en-US", {
    hour: "numeric",
    minute: "numeric",
    second: "numeric",
    hour12: false,
  }).format(date)
);
// > 24:23:16
```
I used both the default locale and configuration that luxon is defining.

**Comment by ramos-ph** [comment]
Changing the locale to either `en-GB`, `de-DE` or `pt-BR` fixes it.
```js
console.log(
  new Intl.DateTimeFormat("en-GB", {
    hour: "numeric",
    minute: "numeric",
    second: "numeric",
    hour12: false,
  }).format(date)
);
// > 00:23:16
```

**Comment by ddimitrioglo** [comment]
@ramos-ph that's a known issue, but I'm pretty sure that luxon maintainers could solve it in 2 ways:
* set `hourCycle: 'h23'` by default
* allow customers to pass `hourCycle` as a parameter based on their needs

By the way, changing locale does not fix it, they just have a different set of default values, that's why for `en-GB`, `de-DE` or `pt-BR` it looks as expected

**Comment by icambron** [comment]
Fixed in 2.0. Thanks for the report
